lib/array_intersection.rb

```diff
@@ -1,6 +1,22 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n x m) where n is the length of array1, and m is the length
+# of array2 since the method loops through both arrays.
+# Space complexity: O(n) because the intersection array stores the output.
+# Other strategies: I was trying to apply binary search to replace the second inner loop
+# but could not get it work because of infinite looping.
 def intersection(array1, array2)
-  raise NotImplementedError
+  return [] if array1.nil? || array2.nil? || array1 == [] || array2 == []
+  return [] if array1 == [] && array2 == []
```

Comment:
line 9 is repetitive - if one of array1 or array 2 == [], an empty array will be returned on line 8